docs/auto-setup.adoc

```diff
@@ -0,0 +1,130 @@
+
+== Installing the system
+
+First, go to the `deployments/local-setup-environment` directory and build the environment image:
+```
+packer build -only=vagrant packerfile.json
+```
+An output directory named `vagrant-box` containing `package.box` file should be created.
+
+Then, switch the directory to `deployments/local-setup-instance` and import
+the image:
+```
+vagrant box add --force --name=local-setup-environment ./../local-setup-environment/vagrant-box/package.box
+```
+Also, install the required `vagrant-disksize` plugin:
+```
+vagrant plugin install vagrant-disksize
+```
+Finally, you can run the virtual machine by doing:
+```
+vagrant up
+```
+A new virtual machine will be created and provisioned accordingly. When
+this command terminates successfully, the machine should be ready to work
+and have all the auxiliary software like Geth, Bitcoin Core, ElectrumX
+and Core/ECDSA clients running. Worth noting, `keep-core` and `keep-ecdsa` client
+executables are always built from their recent master versions.
+
+== How to interact with the system
+
+You can interact with the system through the tBTC dApp or automated
+end-to-end tests. Before you start interacting, you have to connect
+to the virtual machine. First, switch to the `deployments/local-setup-instance`
+directory and do:
+```
+vagrant ssh
+```
+
+You will be moved to the virtual machine home directory where you can find
+the `local-setup` directory where all installed things live.
+
+=== Keep Dashboard dApp
+
+To run the Keep Dashboard dApp invoke:
+```
+./run-keep-dashboard.sh
+```
+
+=== tBTC dApp
+
+To run the tBTC dApp against the local Bitcoin network,
+make sure the tBTC dApp is configured to work with the local ElectrumX instance.
+You can configure this in `./tbtc-dapp/src/wrappers/web3.js` file,
+where `TBTC.withConfig` line occurs. Change it as follows:
+```
+const tbtc = await TBTC.withConfig({
+    web3: web3,
+    bitcoinNetwork: "regtest",
+    electrum: {
+        testnetWS: {
+            server: "127.0.0.1",
+            port: 50003,
+            protocol: "ws"
+        }
+    }
+})
+```
+
+Then, you can invoke:
+```
+./run-tbtc-dapp.sh
+```
+The application will be available on `http://localhost:3000`.
+
+=== E2E tests
+
+To run the automated end-to-end scenario switch to Node 14.3.0:
+```
+nvm use 14.3.0
+```
+Then invoke:
+```
+./run-e2e-test.sh
+```
+
+== Virtual machine lifecycle
+
+You can manage the VM lifecycle by switching to the `deployments/local-setup-instance`
+directory.
+
+To stop the machine do:
+```
+vagrant halt
+```
+
+To start it again, just make:
+```
+vagrant up
+```
+
+To restart the machine, invoke:
+```
+vagrant reload
+```
+
+Everytime you start the machine, Geth, btcd, ElectrumX and Core/ECDSA clients
+will be started as well but the configured `local-setup` directory won't be touched.
+However, if you want to deploy everything from scratch
+you can reload the machine with provisioning option enabled:
+```
+vagrant reload --provision
```

Comment:
Any special steps I need to take?

```
➜  local-setup-instance git:(local-setup-vm) ✗ vagrant reload --provision
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection reset. Retrying...
==> default: Machine booted and ready!
==> default: Checking for guest additions in VM...
    default: The guest additions on this VM do not match the installed version of
    default: VirtualBox! In most cases this is fine, but in rare cases it can
    default: prevent things such as shared folders from working properly. If you see
    default: shared folder errors, please make sure the guest additions within the
    default: virtual machine match the version of VirtualBox you have installed on
    default: your host and reload your VM.
    default:
    default: Guest Additions Version: 5.1.38
    default: VirtualBox Version: 6.1
==> default: Running provisioner: shell...
    default: Running: /var/folders/lk/rwfgns51525745mzgpc0zdhr0000gn/T/vagrant-shell20200723-40501-wtbx2.sh
    default: Cloning repository...
    default: rm: cannot remove 'local-setup/ethereum/data/geth/nodes/LOG': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/nodes/000008.ldb': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/nodes/000005.ldb': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/nodes/LOCK': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/nodes/000006.log': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/nodes/MANIFEST-000007': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/nodes/CURRENT': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/nodes/CURRENT.bak': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/chaindata/ancient/headers.cidx': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/chaindata/ancient/hashes.0000.rdat': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/chaindata/ancient/headers.0000.cdat': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/chaindata/ancient/FLOCK': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/chaindata/ancient/hashes.ridx': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/chaindata/ancient/diffs.0000.rdat': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/chaindata/ancient/diffs.ridx': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/chaindata/ancient/receipts.cidx': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/chaindata/ancient/bodies.cidx': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/chaindata/ancient/receipts.0000.cdat': Permission denied
    default: rm: cannot remove 'local-setup/ethereum/data/geth/chaindata/ancient/bodies.0000.cdat'
    default: : Permission denied
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

Reply:
Please do `vagrant destroy && vagrant up`. Seems like I missed a `sudo` before `./initialize-geth.sh`

Reply:
It worked. I'll try again with just `vagrant reload --provision` once we update scripts.

Reply:
Ok, done. It should work now.

Reply:
Confirmed, works as expected.